skrf/vectorFitting.py

```diff
@@ -45,35 +48,49 @@ def wrapper(*args, **kwargs):
 
 class VectorFitting:
     """
-    =========================================================
-    VectorFitting (:class:`skrf.vectorFitting.VectorFitting`)
-    =========================================================
-
     This class provides a Python implementation of the Vector Fitting algorithm and various functions for the fit
     analysis, passivity evaluation and enforcement, and export of SPICE equivalent circuits.
 
+    Parameters
+    ----------
+    network : :class:`skrf.network.Network`
+            Network instance of the :math:`N`-port holding the frequency responses to be fitted, for example a
+            scattering, impedance or admittance matrix.
+
+    Examples
+    --------
+    Load the `Network`, create a `VectorFitting` instance, perform the fit with a given number of real and
+    complex-conjugate starting poles:
+
+    >>> nw_3port = skrf.Network('my3port.s3p')
+    >>> vf = skrf.VectorFitting(nw_3port)
+    >>> vf.vector_fit(n_poles_real=1, n_poles_cmplx=4)
+
```

Comment:
Question : is it a good practice to put the constructor documentation in the class docstring?

Reply:
Apparently, that is the correct way to di it. See https://developer.lsst.io/python/numpydoc.html#documenting-classes:
```The __init__ method never has a docstring since the class docstring documents the constructor.```
(I did it differently before and the constructor still has its own docstring. So, I still have to fix that)

Reply:
ok, thanks. So I will fix that too for the class I've updated

Reply:
I found various recommendations about docstring and class: 

In the [sphinx extension Napoleon documentation](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html)

>    The __init__ method may be documented in either the class level
    docstring, or as a docstring on the __init__ method itself.
>
>    Either form is acceptable, but the two should not be mixed. Choose one
    convention to document the __init__ method and be consistent with it. 

or advise to put public class attributes in the class documentation. 

[Numpy's Style guide](https://numpydoc.readthedocs.io/en/latest/format.html#documenting-classes) essentially say the same:

> The constructor (__init__) should also be documented here, the Parameters section of the docstring details the constructor’s parameters.

So I guess it's a matter choice. 

As most of the codebase has the `__init__()` already documented, I will not change it for the moment.